torch/csrc/api/include/torch/jit.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <torch/csrc/jit/script/compiler.h>
+#include <torch/csrc/jit/script/module.h>
+#include <torch/csrc/jit/stack.h>
+
+#include <string>
+
+namespace torch {
+namespace jit {
+
+/// Compiles Python JIT code into a graph that can be executed.
+///
+/// For example:
+/// @code
+///   auto module = torch::jit::compile(R"JIT(
+///     def relu_script(a, b):
+///       return torch.relu(a + b)
+///     def test_while(a, i):
+///       while i < 10:
+///         a += a
+///         i += 1
+///       return a
+///   )JIT");
+///   auto output = torch::jit::run(module, "relu_script", a, b);
+///   auto output = torch::jit::run(module, "test_while", a, b);
+/// @endcode
+///
+/// @param source A string containing functions containing script code to
+/// compile
+/// @return A module containing the compiled functions
+std::shared_ptr<script::Module> compile(const std::string& source);
+
+/// Run a method from a module and get a list of the returns.
+///
+/// For example:
+/// @code
+///   auto output = torch::jit::run(module, "relu_script", a, b);
+///   std::cout << output[0].toTensor().toCLong() << std::endl;
+/// @endcode
+///
+/// @param module A module containing the method `method_name` (see
+/// torch::jit::compile)
+/// @param method_name The name of the method to run
+/// @param args Arguments to be passed to the method
+/// @return An IValue containing the return value (or values if it is a tuple)
+/// from the method
+template <typename... Types>
+IValue run(
```

Comment:
If you are adding syntax sugar for running, it should like:
```
auto c = module->run("relu_script", a, b); 
```

It is pretty weird to have this be a method.

Reply:
Agreed, fixed in latest push. It's better on the user end but might get messy later on if we're mixing user-end APIs into the internal JIT code. Should we have a wrapper for `Module` for the C++ API instead?